### ISSUE TITLE
SPARC 'master' build error due to missing Tempus::createStepper()

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#4693]
CC: @trilinos/tempus , @rppawlo (Trilinos Nonlinear Solvers Product Lead), @bartlettroscoe, @fryeguy52, @ccober6 

## Next Action Status

<status-and-or-first-action>

## Description

As shown in [this query](http://compsim-dashboard.sandia.gov/cdash/index.php?project=SPARC&date=2019-03-20&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=66&value1=-trildev&field2=buildstarttime&compare2=83&value2=2019-03-16) the SPARC Trilinos integration builds that build SPARC 'master' against Trilinos 'develop' have a broken build of SPARC where the build error (for example for the `clang-5.0.1` build), shown [here](http://compsim-dashboard.sandia.gov/cdash/viewBuildError.php?buildid=102207)) show:

```
26: error: no matching member function for call to 'createStepper'
    tempus_stepper_ = sf->createStepper(tempus_me_, "Backward Euler");
                     ~~~~^~~~~~~~~~~~~
```

## Current Status on CDash

The status of these tests/builds for the current testing day can be found at:

* [SPARC Trilinos Integration Builds over last 5 days](http://compsim-dashboard.sandia.gov/cdash/index.php?project=SPARC&date=2019-03-20&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=66&value1=-trildev&field2=buildstarttime&compare2=83&value2=5%20days%20ago)

NOTE: The build for the current testing day may not be posted at the time you click this link

## Steps to Reproduce

Instructions for building and installing Trilinos and then building and testing SPARC are given [here](https://snl-wiki.sandia.gov/display/CoodinatedDevOpsATDM/Building+ATDM+APPs+Against+Local+Installs+of+Trilinos#BuildingATDMAPPsAgainstLocalInstallsofTrilinos-BuildingandTestingSPARCAgainstLocalTrilinosInstallation).

To reproduce one will need to be in the SPARC development group to clone the SPARC git repos.

**Comment by rppawlo** [comment]
The model evaluator is now an optional parameter and ordering was swapped so that it could default to null. Just switch the ordering of the string and me.

**Comment by bartlettroscoe** [comment]
@rppawlo said:

> The model evaluator is now an optional parameter and ordering was swapped so that it could default to null. Just switch the ordering of the string and me.

@rppawlo and @ccober6,

Was the PR that caused this non-backward compatible change #4491 which was merged to 'develop' in commit 9d4360d on 3/20/2018?

The problem with just changing SPARC 'master' is that SPARC would need to upgrade Trilinos in order to make this switch because they also need to build against their older accepted version of Trilinos (see [SPARC Trilinos Integration Process](https://snl-wiki.sandia.gov/display/CoodinatedDevOpsATDM/SPARC+Trilinos+Integration+Process).)

Can you provide a deprecated overload with the old ordering?  Otherwise, we have to implement #4532 for which I have not been given the okay to do yet.  Otherwise, SPARC code has no reasonable way to know what version of Trilinos it is building against.

We have to resolve this quickly because this one build error effectively takes down all testing of SPARC against Trilinos 'develop'.

**Comment by micahahoward** [comment]
@bartlettroscoe: thanks for raising this.  I just noticed this yesterday.  What's our process for keeping our Trilinos integration build going until we are able to accept a Trilinos update and make this required change in SPARC/master?

**Comment by bartlettroscoe** [comment]
@micahahoward asked:

> What's our process for keeping our Trilinos integration build going until we are able to accept a Trilinos update and make this required change in SPARC/master?

It is the introduction of an installed `Trilinos_version_date.h` file with the macro `TRILINOS_VERSION_DATE` defined in #4532.   This was the plan we discussed and agreed to outlined in step "4. Manage breaks in backward compatibility in Trilinos" in [SPARC Trilinos Integration Process](https://snl-wiki.sandia.gov/display/CoodinatedDevOpsATDM/SPARC+Trilinos+Integration+Process#SPARCTrilinosIntegrationProcess-Detecting,Triaging,andAddressingNewFailuresinSPARCTrilinosTestingProcess).

I will work on implementing all of this today and try to post a cee-gitlab merge request for SPARC that will fix this and allow SPARC 'master' to build against its current accepted and installed Trilinos version as well as the current Trilinos 'develop'.  I will @mention you in that cee-gitlab MR when it is ready.

**Comment by micahahoward** [comment]
> This was the plan we discussed and agreed to outlined in step "4. Manage breaks in backward compatibility in Trilinos" in SPARC Trilinos Integration Process.

Right.  Thanks for the reminder - couldn't remember how we'd said we'd go about handling backward compatibility issues.

Thanks in advance for posting the fixing MR to cee-gitlab.  We're needing to update SPARC/Trilinos so we'll be working on getting the SPARC/Trilinos integration tests passing.

**Comment by bartlettroscoe** [comment]
FYI: I manually merged the PR #4750 branch to the 'atdm-nightly' branch since the Trilinos PR tester is currently broken and not allowing the merge.  (See commit cbbc5a1).

**Comment by bartlettroscoe** [comment]
This was fixed with the merge of PR #4750 and using the `TRILINOS_VERSION_DATE` macro in SPRAC 'master'.

Closing as complete.
